include/nbl/builtin/hlsl/emulated_float64_t.hlsl

```diff
@@ -0,0 +1,317 @@
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+
+using float32_t = float;
+//using emulated_float64_t = double;
+
+namespace emulated
+{
+    namespace impl
+    {
+        nbl::hlsl::uint32_t2 umulExtended(uint32_t lhs, uint32_t rhs)
+        {
+            uint64_t product = uint64_t(lhs) * uint64_t(rhs);
+            nbl::hlsl::uint32_t2 output;
+            output.x = (product & 0xFFFFFFFF00000000) >> 32;
+            output.y = product & 0x00000000FFFFFFFFull;
+            return output;
+        }

```

Comment:
what do you even need this for?

Reply:
mindlessly copied from mesa, will remove during refactorization

Reply:
all this function does is multiply two 32bit ints to produce a 64bit one and returns MSB in first uint32 and LSB in last

you can rewrite whole of `mul64to128` to work on `uint64_t` instead of pairs of `uint32_t`

e.g. take two `uint64_t` and produce a `uint64_t2` (128 bits)

Reply:
https://stackoverflow.com/a/31662911
see the `umul64wide`

Reply:
i don't need `umulExtendend` anymore